R/clustering.R

```diff
@@ -711,7 +723,7 @@ RunLeiden <- function(
     "matrix" = input <- as(object, "matrix"),
     #run as igraph object (passes to reticulate)
     "igraph" = switch(
-      EXPR = is(object),
+      EXPR = is(object)[1],
       #generate igraph if needed (will handle updated snn class)
       "Graph" = input <- graph_from_adjacency_matrix(adjmatrix = object),
       "dgCMatrix" = input <- graph_from_adjacency_matrix(adjmatrix = object),
```

Comment:
I feel like this whole `switch` construct for the igraph method could be simplified using `inherits`; this way, people can subclass and use their own custom objects and still have it work with `RunLeiden`

```R
input <- if (inherits(x = object, what = 'list')) {
  graph_from_adj_list(adjlist = object)
} else if (inherits(x = object, what = c('dgCMatrix', 'matrix'))) {
  graph_from_adjacency_matrix(adjmatrix = object)
} else if (inherits(x = object, what = 'igraph')) {
  object
} else {
  stop(...)
}
```

Reply:
Complete agree, this is a better approach. I started with this as simple if statement (matrix or igraph) but it's gotten more complex for sparse matrices, lists, etc. I got a bit stuck on how to handle a Seurat "Graph" object (sparse adjmat). It seems to inherit "Matrix" (not "matrix" or "dgCMatrix").

Reply:
I think in either case `graph_from_adjacency` does not require casting to dense to make a `igraph` object.

Reply:
Interesting, the [class definition for `Graph` objects](https://github.com/satijalab/seurat/blob/master/R/objects.R#L153-L156) is simply inheriting from `dgCMatrices` so I wonder why it needs to check inheritance from `Matrix` objects. Either way, doesn't hurt.